javascript/javascript1/week4/appUsage.js

```diff
@@ -0,0 +1,54 @@
+
+//max usage of activity in day
+const usage = [];
+const usageNew = [];
+const durationArr = [];
+
+function addActivity(activity, duration) 
+{
+    const today = new Date();
+    usage.push({
+        date: today.toLocaleDateString("en-US"),
+        activity: activity,
+        duration: duration
+    })
+}
+addActivity("facebook", 10);
+addActivity("twitter", 15);
+addActivity("linked-in", 20);
+addActivity("youtube", 30);
+addActivity("twitter", 25);
+addActivity("whatsapp", 16);
+addActivity("youtube", 20);
+addActivity("twitter", 5);
+console.log(usage);
+
+function calculateActivities() 
+{
+    for (let i = 0; i < usage.length; i++) 
+    {
+        for (let j = i + 1; j < usage.length; j++) 
+        {
+            if (usage[i].activity == usage[j].activity) 
+            {
+                usage[i].duration = usage[i].duration + usage[j].duration;
+                usage.splice(j, 1);
+            }
+        }
+    }
+    let maxkey = "";
+    let maxValue = 0
+    for (let i = 0; i < usage.length; i++) 
+    {
+        if (usage[i].duration > maxValue)
+        {
+            maxkey = usage[i].activity;
+            maxValue = usage[i].duration;
+        }
+
+    }
+    return `The maximum used of ${maxkey} and you used ${maxValue} min`;
+}
+const arr = calculateActivities();
```

Comment:
You are mutating global state. The array 'usage' will now look like this. 

0: {date: "5/29/2021", activity: "facebook", duration: 10}
1: {date: "5/29/2021", activity: "twitter", duration: 45}
2: {date: "5/29/2021", activity: "linked-in", duration: 20}
3: {date: "5/29/2021", activity: "youtube", duration: 50}
4: {date: "5/29/2021", activity: "whatsapp", duration: 16}

your program works, but you have changed the shape of your data, which could have consequences if you needed the data for another function, ie getNumberOfDailyActivities(). 

Can you avoid this?

Reply:
I made another array activityUsage  and push activity, duration.